app/models/user.rb

```diff
@@ -1,2 +1,17 @@
 class User < ApplicationRecord
+  # def recent_posts
+  #   @recent_posts_lists = []
+  #   User.all.each do |each_user|
+  #     each_user.posts.each do |each_post|
+  #      @recent_posts_lists.push(each_user.each_post.title)
+  #     end
+  #   end
+  # end
+  def self.recent_posts(parameters)
+    @recent_posts = Post.limit(3).where(author_id: parameters).order(created_at: :desc)
+  end
```

Comment:
- [ ] Using self keyword when creating a method is redundant. Kindly remove the self keyword.
- [ ] since user has many posts association, you could re-write the code to read:
```
def recent_post
  @recent_posts = posts.order(created_at: :desc).limit(3)
end
```